moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -0,0 +1,355 @@
+#include <map>
+#include <string>
+#include <memory>
+#include <mutex>
+
+#include <eigen3/Eigen/Geometry>
+
+#include <ros/ros.h>
+
+#include <moveit/planning_scene/planning_scene.h>
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit_visual_tools/moveit_visual_tools.h>
+
+#include <tf2_ros/transform_listener.h>
+
+#include <trajectory_msgs/JointTrajectoryPoint.h>
+#include <actionlib/client/simple_action_client.h>
+#include <control_msgs/FollowJointTrajectoryAction.h>
+#include <sensor_msgs/JointState.h>
+
+
+
+namespace moveit_simple
+{
+
+class TrajectoryPoint;
+class JointTrajectoryPoint;
+class CartTrajectoryPoint;
+class Robot;
+typedef std::vector<std::unique_ptr<TrajectoryPoint> > Trajectory;
+
+
+/**
+ * @brief Robot is a wrapper around standard MoveIt objects.  It makes multiple
+ assumptions about the type and complexity of the robot.  Assumptions are:
+ - single arm manipulator (one joint group)
+ - all cartesian poses are of the tool of the robot (could be fixed in the future)
+ - point names are stored as joint positions in the SRDF or frame names in the URDF
+ - motion is executed via a "JointTrajectoryAction" interface
+ - frame transformations outside the URDF are provided by TF
+*/
+class Robot
+{
+public:
+  Robot(const ros::NodeHandle & nh, const std::string &robot_description,
+        const std::string &group_name);
+
+  /**
+   * @brief isInCollision  returns true if pose results in robot config that is
+   * in collision with the environment as defined by the URDF.
+   * @param pose - cartesian pose of the tool to check
+   * @param frame - tool pose relative to frame
+   * @param timeout - (optional) timeout for IK
+   * @param joint_seed (optional) - seed to use
+   * @return
+   */
+  bool isInCollision(const Eigen::Affine3d pose, const std::string & frame,
+                     double timeout = 10.0,
+                     std::vector<double> joint_seed = std::vector<double>() ) const;
+  /**
+   * @brief isReachable - check if point is reacheable
+   * @param name - name of point to check
+   * @param joint_seed - (optional) tries to find joint solutions closest to seed
+   * @param timeout - (optional) timeout for IK
+   * @return
+   */
+  bool isReachable(const std::string & name, double timeout = 10.0,
+                   std::vector<double> joint_seed = std::vector<double>() ) const;
+
+
+  /**
+   * @brief isReachable - check if pose (relative to named frame) is reacheable
+   * @param pose - pose to check
+   * @param frame - frame in which pose is expressed
+   * @param timeout - (optional) timeout for IK
+   * @param joint_seed (optional) - seed to use
+   * @return
+   */
+  bool isReachable(const Eigen::Affine3d & pose, const std::string & frame,
+                   double timeout = 10.0,
+                   std::vector<double> joint_seed = std::vector<double>() ) const;
+
+
+  /**
+   * @brief addTrajPoint - add point to trajectory
```

Comment:
the `addTrajPoint` methods allow you to store trajectories "in" the robot for later execution.  This is more similar to traditional robot programming.  I'd like this class to provide one simple interface that can query robot kinematics and move the robot.  Users of Descartes often struggle with "executing" a trajectory.  It's not clear that they need to define a separate object for this.  
This class will solve that issue.  I realize this isn't the "best" code, since it mixes motion execution and robot kinematics, but this class is meant to collect functionality into a single, easy to use, unit.